### DISCUSSION TITLE
Separated repository for inverter profiles

### DISCUSSION THREAD

**Discussion by davidrapan** [davidrapan/ha-solarman#31] [Ideas]
What about having inverter profiles in separate repository where they can be versioned as standalone and the integration would then use git directly? That would also mean that users could have they own fork of that said repo with their custom changes...

I mean I see only benefits from such feature. What do you people think about it?

Cheers!

**Comment by frsantos** [comment]
In the pros side, we can have the Modbus documentation files with the registers for each inverter model for reference, without this repo being too big.

**Comment by gedger** [comment]
I can see the benefits but I wonder if it would alienate many HA users who struggle to do much more than edit a local YAML file? If this route is taken then I think it would be good to keep the custom directory as is, i.e. keep it for locally managed profiles outside of git. A user is then free to add and manage it via git if they like or just keep it local.

**Reply by davidrapan** [reply]
It would be done in a way which would not affect usability for a standard user. It would stay the same. Git would be used behind the scene. It would just add feature to set your own repo with definition profiles.
